cluster-diagnosis/utils.py

```diff
@@ -131,6 +132,87 @@ def get_pod_config(pod_name):
     return output
 
 
+def get_pods_summarized_status_iterator(pod_name_substring, must_exist=True):
+    """Returns a summarized status of the pods by retrieving the status
+    multiple times.
+
+    This helps avoid any false negatives that can occur
+    in the scenario wherein the status is checked just after a pod restart.
+
+    Args:
+        pod_name_substring - the substring containing the pod name.
+        must_exist - boolean to indicate that a pod with that name must exist.
+            If the condition isn't satisfied, an error will be logged.
+
+    Returns:
+        name (string): name of the pod.
```

Comment:
Returns:
A tuple with the following elements in this order:
...

Reply:
You could also define a namedtuple and use it as return type to better document the structure and make the callers more readable.